hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/AlterHoodieTableAddColumnsCommand.scala

```diff
@@ -110,15 +110,15 @@ object AlterHoodieTableAddColumnsCommand {
       HoodieWriterUtils.parametersWithWriteDefaults(hoodieCatalogTable.catalogProperties).asJava
     )
 
-    val commitActionType = CommitUtils.getCommitActionType(WriteOperationType.INSERT, hoodieCatalogTable.tableType)
+    val commitActionType = CommitUtils.getCommitActionType(WriteOperationType.ALTER_SCHEMA, hoodieCatalogTable.tableType)
```

Comment:
does this work even if the writer does not turn on schema evolution originally for the table? say it was created using DeltaStreamer tool

Reply:
maybe i missed somethings， could you pls explain it in detail
now DDL operation is only support by sparksql，  if we does not turn on schema evolution originally for the table， it donot work.
sorry  I'm not sure if this explanation is what you want